### PR TITLE
raftstore: improve the remove peer check

### LINKED ISSUE #16465: (dr-autosync) v6.5.8 scale in one node from primary datacenter failed during async mode
### What version of TiKV are you using?

v6.5.8

### Steps to reproduce

1. deploy one tidb cluster with 12 stores, and config it with 5 replicas in dr-auto-sync mode.
2. down two tikvs in the backup center, make replication mode switch to async.
3. Try to scale-in one node in the primary center.

### What did you expect?
Scale in succeeded.

### What did happened?
Scale in hung.

Didn't hit it before https://github.com/tikv/tikv/pull/16174.

### PR DESCRIPTION
**Pull request by tonyxuqqi**
### What is changed and how it works?

Issue Number: Close #16465

What's Changed:

```commit-message
improve the remove peer check. Only check when the updating role is voter
```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by LykxSassinator — APPROVED** [review]

File: components/raftstore/src/store/util.rs [hunk 1]
```diff
@@ -1139,16 +1133,24 @@ fn check_remove_or_demote_voter(
...
-        if *id == leader_id || last_heartbeat.elapsed() <= slow_peer_threshold {
-            normal_peer_count += 1;
-        } else {
-            slow_peer_count += 1;
+        // for slow and normal peer calculation, we only count voter role
+        if region
+            .get_peers()
+            .iter()
+            .find(|p| p.get_id() == *id)
+            .map_or(false, |p| p.role == PeerRole::Voter)
```
Comment (Connor1996) [inline components/raftstore/src/store/util.rs:1146]: What about IncomingVoter and DemotingVoter

Comment (tonyxuqqi) [reply components/raftstore/src/store/util.rs:1146]: check_remove_or_demote_voter is called only when it's simple or EnteringJoint, which means the role could never be IncomingVoter or DemotingVoter before the raftcmd is executed. @Connor1996

**Review by Connor1996 — APPROVED** [review]
LGTM

File: components/raftstore/src/store/util.rs [hunk 2]
```diff
@@ -1158,10 +1160,16 @@ fn check_remove_or_demote_voter(
         if change_type == ConfChangeType::RemoveNode
             || change_type == ConfChangeType::AddLearnerNode
         {
+            let is_voter = region
```
Comment (v01dstar) [inline components/raftstore/src/store/util.rs:1163]: Recommend renaming `normal_peers_to_remove` and `slow_peers_count` to `normal_voters_to_remove` and `slow_voters_count`

Comment (v01dstar) [reply components/raftstore/src/store/util.rs:1163]: #16476 16476

**Pull request merged**